addons/mrp/models/mrp_workorder.py

```diff
@@ -783,7 +790,7 @@ def _get_duration_expected(self, alternative_workcenter=False, ratio=1):
             if duration_expected_working < 0:
                 duration_expected_working = 0
             return self.workcenter_id._get_expected_duration(self.product_id) + duration_expected_working * ratio * 100.0 / self.workcenter_id.time_efficiency
-        qty_production = self.production_id.product_uom_id._compute_quantity(self.qty_production, self.production_id.product_id.uom_id)
+        qty_production = self.production_id.product_uom_id._compute_quantity(self.qty_produced or self.qty_producing or self.qty_production, self.production_id.product_id.uom_id)
```

Comment:
is there a case where `qty_produced` is not 0 but also different from `qty_producing` ? Because if not, you can just use `qty_producing`

Reply:
If I am not mistaken when the workorder is done the `qty_producing` is 0 while the `qty_produced` is full. I have not checked when this switch happens, but as such, it felt safer to use both quantities as it  provides a reliable result no matter when the computation is done or how this switch is done.